### PR TITLE
Ensure password-less users can change email

### PR DESCRIPTION
**Pull request by dcramer**
@getsentry/infrastructure

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/3674)

### CODE REVIEW

File: src/sentry/web/forms/accounts.py [hunk 1]
```diff
@@ -264,24 +264,26 @@ def __init__(self, user, *args, **kwargs):
         self.user = user
         super(AccountSettingsForm, self).__init__(*args, **kwargs)
 
+        needs_password = bool(user.password)
```
Comment (mitsuhiko) [inline src/sentry/web/forms/accounts.py]: I have seen user password being set to `!`. Is that comparable?

**Comment by dcramer** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/sentry/web/forms/accounts.py, line 267 \[r1\]](https://reviewable.io:443/reviews/getsentry/sentry/3674#-KMBATj-AQ1n8A0dPxCp-r1-267:-KMBBded5s0Ic5XjnV-G:-315073917) ([raw file](https://github.com/getsentry/sentry/blob/f448e6891951e7b6ccb94dc7f355f3bfa2c27f38/src/sentry/web/forms/accounts.py#L267)):*
<details><summary><i>Previously, mitsuhiko (Armin Ronacher) wrote…</i></summary>
> I have seen user password being set to `!`. Is that comparable?
</details>
Found a Django API which covers it

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3674)*

**Comment by mattrobenolt** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3674)*

**Comment by mitsuhiko** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3674)*

**Pull request merged**